### PR TITLE
Views for keyword router

### PR DESCRIPTION
**Pull request by jerith**
This is the part of #277 that needs to live in `develop-frontend`.

### CODE REVIEW

**Comment by jerith** [comment]
This isn't finished yet, but I'd like a review on the bits I've done so far.

File: go/base/utils.py [hunk 1]
```diff
@@ -131,3 +143,18 @@ def get_conversation_view_definition(conversation_type, conv=None):
                 return ConversationViewDefinitionBase(conv_def)
             return app_pkg.view_definition.ConversationViewDefinition(conv_def)
     raise Exception("Can't find conversation_type: %s" % (conversation_type,))
+
+
+def get_router_view_definition(router_type, router=None):
+    # Scoped import to avoid circular deps.
+    from go.router.view_definition import RouterViewDefinitionBase
+    # HACK: Assume 'namespace' is 'router_type'.
```
Comment (hodgestar) [inline go/base/utils.py:151]: Stll not sure this is a hack. Maybe let's just document them as the same thing? There's probably room to clear this up somehow (i.e. put the name as a constant in the module or something).

Comment (hodgestar) [inline go/base/utils.py:151]: Not for this PR -- just reminding us.

File: go/base/utils.py [hunk 2]
```diff
@@ -131,3 +143,18 @@ def get_conversation_view_definition(conversation_type, conv=None):
...
+    for module, data in settings.VUMI_INSTALLED_ROUTERS.iteritems():
+        if data['namespace'] == router_type:
+            app_pkg = __import__(module,
```
Comment (hodgestar) [inline go/base/utils.py]: Maybe `router_pkg`?

File: go/router/templates/router/dashboard.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+{% block content_extraclass %}dashboard routers{% endblock %}
+
+{% block content_title %}All routers{% endblock %}
+
+
+{% block content_actions_right %}
+    <div class="table-form-view-buttons pull-left">
+        <button class="btn" data-action="archive" disabled="disabled">Release</button>
```
Comment (hodgestar) [inline go/router/templates/router/dashboard.html]: s/Release/Archive/ ?

File: go/router/templates/router/new.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{% extends "app.html" %}
+
+{% block content_title %}New Router{% endblock %}
+
+{% block content_main %}
+<div class="row">
+    <div class="span12">
+        {% include "base/includes/messages.html" %}
+    </div>
+</div>
```
Comment (hodgestar) [inline go/router/templates/router/new.html:10]: @peterp is planning to put messages in a different content block so we don't have to duplicate this everywhere.

**Comment by hodgestar** [comment]
Looking good so far!

**Comment by hodgestar** [comment]
Other than the suggestion to more neatly update the set of endpoints and to perhaps make `default` less special by just putting it in the default static endpoint lists, :+1:.

I have a question about whether we need an option to turn off regexes (and probably also a case-sensitive / not option) but that's for another PR, I think.

**Comment by jerith** [comment]
I'll change the name of the edit fields. We can change the behaviour in develop.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**